misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -296,6 +297,13 @@ ClientConfigHolder getClientConfigHolder() {
                     .filter(JBallerinaDebugServer::isValidFrame)
                     .toArray(StackFrame[]::new);
 
+            if (context.getLastInstruction() == DebugInstruction.STEP_IN
+                && validFrames[0].getSource().getPath().equals(topStackFrame.getSource().getPath())
```

Comment:
Potential `IndexOutOfBoundsException` as the `validFrames` can be an empty array

Reply:
Fixed